src/doc/rustc/src/codegen-options/index.md

```diff
@@ -416,11 +439,53 @@ See also the [`target_feature`
 attribute](../../reference/attributes/codegen.md#the-target_feature-attribute)
 for controlling features per-function.
 
-This also supports the feature `+crt-static` and `-crt-static` to control
-[static C runtime linkage](../../reference/linkage.html#static-and-dynamic-c-runtimes).
+#### Static linking
 
-Each target and [`target-cpu`](#target-cpu) has a default set of enabled
-features.
+Target feature `crt-static` can be used to enable or disable static linking,
+with the default depending on the current target.  
+`-C target-feature=+crt-static` is similar to `-static` in other compilers,
+standard C library in particular is linked statically in this case.
+
+The exact meaning of `+crt-static` depends heavily
+on the current target and the produced crate type.
+
+- ELF-based executables.  
+If the target supports statically linked executables,
+then the linker will be instructed (`-static`) to produce one.  
+The executable will be self-contained,
+will contain code from all the used libraries, including `libc`, inside it,
+and will be able to run without a dynamic loader (no `INTERP` header).  
+
+    If the target doesn't support both position-independent and statically linked executables,
+then `-C target-feature=+crt-static` "wins" over `-C relocation-model=pic`,
+and the linker is instructed (`-static`) to produce a statically linked
+but not position-independent executable.
+
+- ELF-based shared libraries.  
+If the target supports statically linked shared libraries,
+then the linker will be instructed (`-shared -static`) to produce one.
+The shared library will contain code from all the used libraries, including `libc`, inside it.
+This option is not widely supported or used.
+
+- PE-based executables (Windows).  
+The produced executable will contain code from all the user-level libraries, including
+[C Run-Time Libraries (CRT)](https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features),
```

Comment:
What user-level libraries will it statically link to other than the CRT? For `windows-msvc` `crt-static` *only* controls whether the CRT (consisting of the VC Runtime and the UCRT) is statically linked and nothing else is affected. Also it only applies to `windows-msvc` while `windows-gnu` does not support `crt-static`.

Reply:
@retep998 
>What user-level libraries will it statically link to other than the CRT?

Rust crates.
And it should do that for native dependencies as well (https://github.com/rust-lang/rust/issues/71647), but that doesn't make any difference on Windows since there's no `libmylib.a`/`libmylib.so` separation.

If after that anything is still linked dynamically, then we get into the `*WARNING!*` clause below. It's not as dangerous in the Windows case though, since the goal there is not to produce a fully statically linked executable, like in the ELF case.

>Also it only applies to `windows-msvc` while `windows-gnu` does not support `crt-static`.

Good catch.
This paragraph needs the same "If the target supports..." condition as the ELF paragraphs above.

Reply:
To me `crt-static` on Windows always had the impression that the only thing it controlled (and ever intended to control) was whether it linked to the static or dynamic CRT. I was not aware that it had any effect on whether a rust crate was linked statically or dynamically, as `-Cprefer-dynamic` already covers that.